src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -369,38 +404,43 @@ public class SqlMutationEngine : IMutationEngine
                     throw _dabExceptionWithTransactionErrorMessage;
                 }
 
-                DbResultSetRow? dbResultSetRow = upsertOperationResult is not null ?
-                    (upsertOperationResult.Rows.FirstOrDefault() ?? new()) : null;
+                Dictionary<string, object?> resultRow = upsertOperationResultSetRow!.Columns;
 
-                if (upsertOperationResult is not null &&
-                    dbResultSetRow is not null && dbResultSetRow.Columns.Count > 0)
+                // For MsSql, MySql, if it's not the first result, the upsert resulted in an INSERT operation.
+                // Even if its first result, postgresql may still be an insert op here, if so, return CreatedResult
```

Comment:
nit: remove return CreatedResult; only the upsert resulted in an update/insert is being determined